client/consensus/src/collators.rs

```diff
@@ -284,28 +284,66 @@ impl<Pub: Clone> SlotClaim<Pub> {
 }
 
 /// Attempt to claim a slot locally.
-pub fn tanssi_claim_slot<P>(
-    authorities: Vec<AuthorityId<P>>,
+pub fn tanssi_claim_slot<P, B>(
+    aux_data: OrchestratorAuraWorkerAuxData<P>,
+    chain_head: &B::Header,
     slot: Slot,
     force_authoring: bool,
     keystore: &KeystorePtr,
 ) -> Result<Option<SlotClaim<P::Public>>, Box<dyn Error>>
 where
-    P: Pair,
+    P: Pair + Send + Sync + 'static,
     P::Public: Codec + std::fmt::Debug,
     P::Signature: Codec,
+    B: BlockT,
 {
     let author_pub = {
-        let res = claim_slot_inner::<P>(slot, &authorities, keystore, force_authoring);
+        let res = claim_slot_inner::<P>(slot, &aux_data.authorities, keystore, force_authoring);
         match res {
             Some(p) => p,
             None => return Ok(None),
         }
     };
 
+    if is_parathread_and_should_skip_slot::<P, B>(&aux_data, chain_head, slot) {
+        return Ok(None);
+    }
+
     Ok(Some(SlotClaim::unchecked::<P>(author_pub, slot)))
 }
 
+/// Returns true if this container chain is a parathread and the collator should skip this slot and not produce a block
+pub fn is_parathread_and_should_skip_slot<P, B>(
+    aux_data: &OrchestratorAuraWorkerAuxData<P>,
+    chain_head: &B::Header,
+    slot: Slot,
+) -> bool
+where
+    P: Pair + Send + Sync + 'static,
+    P::Public: Codec + std::fmt::Debug,
+    P::Signature: Codec,
+    B: BlockT,
+{
+    if slot.is_zero() {
+        // Always produce on slot 0 (for tests)
+        return false;
+    }
+    if let Some((start, _end)) = aux_data.next_slot_range_inclusive {
+        if let Ok(chain_head_slot) = find_pre_digest::<B, P::Signature>(chain_head) {
+            let slot_diff = slot.saturating_sub(chain_head_slot);
+
+            // TODO: this ignores force authoring
```

Comment:
what does this sentence mean?

Reply:
That even if we set force_authoring to true, this collator will not propose blocks on every slot